### ISSUE TITLE
Stars hidden even on mouseover of empty space; and remain hidden whilst right click menu is open

### ISSUE THREAD

**Issue by mxmilkiib** [mixxxdj/mixxx#12576]
### Bug Description

Reanalysing a track also resets the rating.

I just went down two or three wrong steps in trying to 3/2 or 4/3 or whatever the BPM and got lost (especially as I've a bad working memory), and I want to reset the BPM, and ok if the key and waveform is redone at the same time; but when I think "Reanalyse" in this context, I don't think "reset a tracks rating". I would think this would be a reanalysis of the "techinical" understanding/components of the track, not my understanding of whether I like it or not.

I'm guessing the majority of folk would have the same take as me on this.

### Version

2.5-alpha

### OS

Linux

**Comment by ronso0** [comment]
This does not happen here, neither with 2.4 nor 2.5-alpha.
I used the track context menu in the tracks table > Analyze > Reanalyze.
Having the track loaded in a deck doesn't make a difference.

How did you trigger analysis?

**Comment by mxmilkiib** [comment]
I know what it is;

tl;dr
* Make the rating field a bit longer than the stars take up.
* Hover over that extra bit of field; the stars get hidden
* Right click, go into menu, stars still remain hidden
* Sometimes the [sub]menu remains open for longer than the regular timeout (or clicking left/right/hitting esc, these I all managed to confirm are closing actions before I lost the ability/happenstance to reproduce long menus)

I the last few times I so happen to have (because IDK the stars have been my main target; it's just so happens to feel good, gestalt,, because it's in the "middle" and "not taken up", even though it doesn't matter where the right click is) on the space just to the right of the stars.

And when you right click and go into the menu to do whatever, the state of 0 stars remains, and then when one moves the mouse off of the menu or submenu the menu remains open;

Even with just a small extra space (cursor not over space because it would just show no stars, and I forgot);
![image](https://github.com/mixxxdj/mixxx/assets/108225/5e377442-5aef-42e7-aaf0-f121bbc8de2e)

And sometimes the menu/submenus remain open for a fair while after moving the mouse off, sometimes  they not.  There was a time out of like 1.5s when I tried to duplicate, but I've had instances where I move the mouse off to a certain spot and the menu has stayed open much longer than that.

Kinda similar to #10110

**Comment by ronso0** [comment]
Yes, I can confirm this, and it's indeed confusing.
Something I _tried_ to fix #11772. (didn't work)
Just now I took another look, et voilà #12582
